synapse/event_auth.py

```diff
@@ -69,10 +69,11 @@ def check(
     # sanity-check it
     for auth_event in auth_events.values():
         if auth_event.room_id != room_id:
-            raise Exception(
+            raise AuthError(
+                500,
```

Comment:
shouldn't this be a 403?

Reply:
oh oof there is AuthError(500) above. whatever.

Reply:
Oh, yes, probably actually. I was copying it from a previous line, but I don't think that makes sense

Reply:
but also: if we actually hit this code and it matters, the comment needs changing.

Reply:
/me awards past me a prize for including sanity checks for security-critical things

Reply:
Sorry, which comment?

Reply:
Found it